src/clj/witan/send/goal_seek.clj

```diff
@@ -25,35 +31,40 @@
                       (update :population i/->int)))
             filepath))
 
-(defn state-pop
-  "Takes a config and returns the state to be modified and output path of
-   Output_State_pop_only.csv results"
-  [config]
-  (let [state (-> (get-in config [:transition-parameters :transitions-to-change])
-                  first
-                  (clojure.set/rename-keys {:setting-2 :setting :need-2 :need :academic-year-2 :academic-year})
-                  (select-keys [:setting :need :academic-year]))
-        out-dir (:output-dir (:output-parameters config))
-        state-pop (str (:project-dir config)
-                       "/"
-                       out-dir
-                       "/Output_State_pop_only.csv")]
-    [state state-pop]))
+(defn match-need-setting [data state]
+  (= (second (key data)) (st/join-need-setting (:need state) (:setting state))))
+
+(defn match-ay [data state]
+  (let [ay-state (:academic-year state)]
+    (if (coll? ay-state)
+      (true? (some (fn [ay] (= (first (key data)) ay)) ay-state))
```

Comment:
Does this need `true?` or will truthy do?

Reply:
Had to debug and check, but yes because we want at least one (hence `some`) of `(first (key data))` to equal `ay` this is actually testing for `true?`